src/compiler/checker.ts

```diff
@@ -5885,22 +5888,20 @@ module ts {
         function inferTypeArguments(signature: Signature, args: Expression[], excludeArgument?: boolean[]): InferenceContext {
             var typeParameters = signature.typeParameters;
             var context = createInferenceContext(typeParameters, /*inferUnionTypes*/ false);
-            var mapper = createInferenceMapper(context);
-            // First infer from arguments that are not context sensitive
+            var inferenceMapper = createInferenceMapper(context);

```

Comment:
Technically, you may not use the inference mapper at if all your arguments are sensitive to contextual typing. This was already the case. I doubt it's a worthwhile optimization, but I'm just thinking out loud (in writing).